File: 2026/08-August/19-Wednesday/solution.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public int MaxNumberOfFamilies(int n, int[][] reservedSeats)
    {
        Dictionary<int, HashSet<int>> reserved = new();

        foreach (int[] seat in reservedSeats)
        {
            if (!reserved.ContainsKey(seat[0]))
                reserved[seat[0]] = new HashSet<int>();

            reserved[seat[0]].Add(seat[1]);
        }

        int ans = (n - reserved.Count) * 2;

        foreach (HashSet<int> seats in reserved.Values)
        {
            bool left = !seats.Contains(2)
                     && !seats.Contains(3)
                     && !seats.Contains(4)
                     && !seats.Contains(5);

            bool middle = !seats.Contains(4)
                       && !seats.Contains(5)
                       && !seats.Contains(6)
                       && !seats.Contains(7);

            bool right = !seats.Contains(6)
                      && !seats.Contains(7)
                      && !seats.Contains(8)
                      && !seats.Contains(9);

            if (left && right)
                ans += 2;
            else if (left || middle || right)
                ans += 1;
        }

        return ans;
    }
}

File: 2026/08-August/19-Wednesday/solution.java
import java.util.*;

class Solution {
    public int maxNumberOfFamilies(int n, int[][] reservedSeats) {
        Map<Integer, Set<Integer>> reserved = new HashMap<>();

        for (int[] seat : reservedSeats) {
            reserved
                .computeIfAbsent(seat[0], k -> new HashSet<>())
                .add(seat[1]);
        }

        // Rows without any reserved seats can fit 2 families.
        int ans = (n - reserved.size()) * 2;

        for (Set<Integer> seats : reserved.values()) {
            boolean left = !seats.contains(2)
                    && !seats.contains(3)
                    && !seats.contains(4)
                    && !seats.contains(5);

            boolean middle = !seats.contains(4)
                    && !seats.contains(5)
                    && !seats.contains(6)
                    && !seats.contains(7);

            boolean right = !seats.contains(6)
                    && !seats.contains(7)
                    && !seats.contains(8)
                    && !seats.contains(9);

            if (left && right) {
                ans += 2;
            } else if (left || middle || right) {
                ans += 1;
            }
        }

        return ans;
    }
}
